tooling/build/scripts/publisher.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# Helper function to calculate duration
+calculate_duration() {
+  start_time=$1
+  end_time=$(date +%s)
+  duration=$((end_time - start_time))
+  echo "Time taken: $duration seconds"
+}
+
+# Cloning the repository
+echo "Cloning repository..."
+start_time=$(date +%s)
+
+# TODO: add single branch checkout with --branch main
+git clone --depth 1 --branch 08-03-add_publishing_scripts https://github.com/opengovsg/isomer.git
+cd isomer/
+calculate_duration $start_time
+
+echo $(pwd)
+
+# Checkout specific branch
+echo "Checking out branch..."
+start_time=$(date +%s)
+git checkout 08-03-add_publishing_scripts
+calculate_duration $start_time
+
+echo $(git branch)
+
+# Perform a clean of npm cache
+# Example of error without this: 
+# npm ERR! EEXIST: file already exists, open '/root/.npm/_cacache/tmp/7746bf9d'
+npm cache clean --force
+
+# Install dependencies
+echo "Installing dependencies..."
+start_time=$(date +%s)
+npm ci
+calculate_duration $start_time
+
+# Build components
+echo "Building components..."
+start_time=$(date +%s)
+cd packages/components
+npm run build
+cd ../.. # back to root
+echo $(pwd)
+calculate_duration $start_time
+
+# Fetch from database
+echo "Fetching from database..."
+start_time=$(date +%s)
+cd tooling/build/scripts/publishing
+echo $(pwd)
+npm ci
+npm install ts-node -g
+npm run start
+calculate_duration $start_time
+
+# Build site
+echo "Building site..."
+start_time=$(date +%s)
+rm -rf ../../../template/schema
+mv schema/ ../../../template/
+cd ../../../template
+echo $(pwd)
+npm ci
+
+# Prebuild
+echo "Prebuilding..."
+rm -rf node_modules && rm -rf .next
+curl https://raw.githubusercontent.com/opengovsg/isomer/main/tooling/build/scripts/preBuild.sh | bash
+
+# Build
+echo "Building..."
+rm -rf scripts/
+curl https://raw.githubusercontent.com/opengovsg/isomer/main/tooling/build/scripts/build.sh | bash
```

Comment:
nit: personal preference but I think we can combine the preBuild and build scripts into this one so it is easier to update/test in the future.

Reply:
makes sense, made the changes